magento/pwa-studio
# PWA-2311: Update Breadcrumb.category_id GraphQL field to category_uid

- Updated graphql query and accompanying tests



## Description

Overview
The field `Breadcrumb.category_id` is deprecated. We should use the `category_uid` field instead. 

Found in:

peregrine/lib/talons/Breadcrumbs/breadcrumbs.gql.js
Acceptance Criteria
PWA Studio storefronts no longer use the `Breadcrumb.category_id` deprecated field

## Related Issue







[PWA-2311](https://jira.corp.magento.com/browse/PWA-2311)

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






Test Plan
Navigate to a page that uses breadcrumbs in Venia (any category landing page)
Find the "GetBreadcrumbs" call to GraphQL in the browser network tab
In the "Payload" tab, click "view decoded"
Verify that we are not requesting the `category_id` field in the `breadcrumbs` response
Test upgrade flow. So either on cloud or local hit all the changed area above in 12.1 (master)  and then run staging server with this pr and check all areas in new tab of same browser. Details steps on upgrad test can be found here https://jira.corp.magento.com/browse/PWA-1418
 This is bad:

{breadcrumbs{category_id category_level category_name category_url_path __typename}}
 This is good:

{breadcrumbs{category_uid category_level category_name category_url_path __typename}} 

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
